src/sentry.ts

```diff
@@ -0,0 +1,33 @@
+import * as SentryRenderer from '@sentry/electron/renderer'
+import { AppSettings } from './types/settings'
+
+export function configureRendererProcess(
+  getSettingsFn?: () => Promise<AppSettings>
+) {
+  if (process.env.NODE_ENV !== 'development') {
+    SentryRenderer.init({
+      integrations: [
+        SentryRenderer.browserTracingIntegration(),
+        SentryRenderer.replayIntegration(),
+      ],
+      tracesSampleRate: 1.0,
+      replaysSessionSampleRate: 0.1,
+      replaysOnErrorSampleRate: 1.0,
+
+      // conditionally send the event based on the user's settings
+      beforeSend: async (event) => {
+        const getSettings = getSettingsFn || window.studio.settings.getSettings
```

Comment:
`window` isn't available in the preload context so I'm adding flexibility here to pass the getSettings function as an argument optionally.